src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/SpecialFileProviders/AbstractSpecialFileProvider.cs

```diff
@@ -99,7 +99,9 @@ internal abstract class AbstractSpecialFileProvider : ISpecialFileProvider
             // We haven't found the file but return the default file path as that's the contract.
             IProjectTree rootNode = _projectTreeService.CurrentTree.Tree;
             string rootFilePath = _projectTreeService.CurrentTree.TreeProvider.GetPath(rootNode);
-            string fullPath = Path.Combine(rootFilePath, specialFileName);
+            string fullPath = Path.Combine(
+                rootNode.IsFolder ? rootFilePath : Path.GetDirectoryName(rootFilePath),
```

Comment:
I don't know of any case where it would be. If we are absolutely sure then we can remove that condition